src/main/java/org/dimdev/accesstransform/AccessTransformationSet.java

```diff
@@ -1,17 +1,70 @@
 package org.dimdev.accesstransform;
 
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Map;
-import java.util.Set;
+
+import org.dimdev.accesstransform.ElementReference.Kind;
 
 public class AccessTransformationSet {
-    private final Map<ElementReference, AccessLevel> transformations = new HashMap<>();
-    private final Set<String> affectedClasses = new HashSet<>();
+	public static class Transformation {
+		final Map<ElementReference, AccessLevel> methods = new HashMap<>();
+		final Map<ElementReference, AccessLevel> fields = new HashMap<>();
+		private final String name;
+		AccessLevel access;
+
+		public Transformation(String name) {
+			this.name = name;
+		}
+
+		public boolean wantsAccessChange() {
+			return access != null;
+		}
+
+		public AccessLevel getAccessChange() {
+			return access;
+		}
+
+		public boolean wantsMethodChange() {
+			return !methods.isEmpty();
+		}
+
+		public AccessLevel popMethod(String name, String description) {
+			return methods.remove(new ElementReference(Kind.METHOD, this.name, name, description));
+		}
+
+		public boolean wantsFieldChange() {
+			return !fields.isEmpty();
+		}
+
+		public AccessLevel popField(String name, String description) {
+			return fields.remove(new ElementReference(Kind.FIELD, this.name, name, description));
+		}
+
+		public void ensureClear() {
+			if (!methods.isEmpty() || !fields.isEmpty()) {
+				throw new IllegalStateException("Additional transformations to " + name + " missed: methods: " + methods + ", fields: " + fields);
+			}
+		}
+	}
+    private final Map<String, Transformation> transformations = new HashMap<>();
 
     public void addMimimumAccessLevel(ElementReference elementReference, AccessLevel accessLevel) {
-        transformations.put(elementReference, AccessLevel.union(transformations.get(elementReference), accessLevel));
-        affectedClasses.add(elementReference.kind == ElementReference.Kind.CLASS ? elementReference.name : elementReference.owner);
+    	switch (elementReference.kind) {
+    		case CLASS: {
+    			Transformation transform = transformations.computeIfAbsent(elementReference.name, Transformation::new);
+    			transform.access = AccessLevel.union(transform.access, accessLevel);
+    		}
```

Comment:
Shouldn't there be a break at the end of these cases to prevent case falling and leading to issues?

Reply:
Oops, there should indeed